sorted_db/db.go

```diff
@@ -136,21 +136,17 @@ func indexByte(s []byte, i, m int, c byte) int {
 	return -1
 }
 
-// Search uses a binary search looking for needle, and returns the full match line.
-// the needle should already have the record separator appended
-func (db *DB) Search(needle []byte) []byte {
-	db.RLock()
-
+// findFirstRecordAfterMatch performs a binary search to find the first record after
+// needle prefix matches. If includeExactMatch is set, returns the first record that exactly matches,
+// if it exists.
+func (db *DB) findFirstRecordAfterMatch(needle []byte, includeExactMatch bool) int {
 	needleLen := len(needle)
 
-	if db.size <= 0 {
-		panic("DB not Mapped")
-	}
 	// binary search to find the index that matches our needle (starting at the previous line)
 	// note: this could be more efficient if we wrote our own search as we could skip data we've checked
 	// isntead of checking potentially more indexes here. Because page sizes is 4k this should hopefully
```

Comment:
s/isntead/instead/

also:

s/Because page sizes is 4k/Because page sizes are 4k/